### ISSUE TITLE
[Issue] Image metadata ends up with JSON instead of string

### ISSUE THREAD

**Issue by vladmandic** [vladmandic/automatic#1356]
### Discussed in https://github.com/vladmandic/automatic/discussions/1355

<div type='discussions-op-text'>

<sup>Originally posted by **Tillerz** June  8, 2023</sup>
Hi,

is this only me?

I am on version: 276fb6a0 Wed Jun 7 17:45:39 2023 -0400

When I load an image into "process image" and then click "--> Text", it copies all the settings, but it deletes the prompt and negative prompt, not filling in the data from the image.

**Comment by Tillerz** [comment]
Still happenes after reinstalling from scratch. Olly things I copied over were output/ and models/.

**Comment by Tillerz** [comment]
Here a sample image with the JSON header:

[sample-image.zip](https://github.com/vladmandic/automatic/files/11688293/sample-image.zip)

**Comment by vladmandic** [comment]
ok, i'm going crazy now. i've tried fresh clone and install (so everything new, even venv)  
and all possible combos of generating new image in txt2img (with and without hires) or upscale in process and i cannot reproduce.

and yes, i can see that your image has string-encoded-json in exifmetadata.
you don't need a hex dump, i prepackage a tool to view any image raw metadata (not just from this app) - `cli/image-exif.py`

**Comment by vladmandic** [comment]
so really treat me like an idiot and step by step how do you generate image that has such metadata?
don't focus on reading/parsing, that's sideeffect of metadata in the file being wrong to start with.

**Comment by TheOnlyHolyMoly** [comment]
Is that a an image that @Tillerz generated or an external image?

**Comment by Tillerz** [comment]
I can just open SD, type a prompt, click generate, save it (via save button). It has JSON in it. Nothing else. Just that.

**Comment by vladmandic** [comment]
> I ALWAYS only use save, not the automatic stuff, because I don't need all the broken things that get generated inbetween, that's why I have 'save all generated images' off. Always.

that is pretty CRITICAL piece of information
i did ask 5 messages ago what are your image options as i strongly suspected it had something to do how image is saved to start with!
and you did state explicitly the issue is reproducible with fresh install and default config. well, that setting is NOT default.

so i wasted time trying to reproduce the wrong way.

ok, let me try again.

**Comment by Tillerz** [comment]
The problem is not the toggle. The problem is behind the SAVE button.

I just tried to verify where it fucks up, and it does that (it seems) only on for the save/ images created by clicking that save button, the automatic saves seem to be ok.

**Comment by vladmandic** [comment]
i asked the question for a reason, not you're assuming it doesn't matter.
if file is auto-saved, save does not run through save at all, it runs a file copy operation.

**Comment by Tillerz** [comment]
Sorry, wasn't assuming, I just tried to thing of what you differently from what I do, then I had the idea that you maybe look at the images it saves itself, which I do not, so I enabled that to validate that output. :)

**Comment by vladmandic** [comment]
this is much worst than you could even imagine.

when image is auto-saved, its done at the end of generate process and it has access to all processing information/params.
and when you use save button later, image is copied as there is no need to run any kind of processing again.
i've redone both a while back and it works pretty well.

when image is NOT auto-saved and you use save button, where does any kind of info come from?
original processing info is long gone, so it does that by doing gymnastics of copying and combining what is already printed in ui in different text boxes.
all because a1111 never had a concept of result OBJECT.

and yes, i broke that - because i was trying to unify entire image gallery button bar to be uniform between txt2img/img2img/extras tabs - previously each one had its own special handlers.

look at this **ugly** diff to fix it this: <https://github.com/vladmandic/automatic/commit/c82eaef261b6717599e27068d19a4b5425c66aec>

anyhow, updated.

**Comment by Tillerz** [comment]
Glad you found it, woohoo.

Oi, that's a lot of changes to fix it. :-o

```
original processing info is long gone, so it does that by doing 
gymnastics of copying and combining what is already printed in
ui in different text boxes.
```

Does that also mean when I edit the parameters and then find out I didn't save the previous image and click save, it saves it with the wrong (newly edited) data? :-o

**Comment by vladmandic** [comment]
no, it ignores params, it looks at text boxes below gallery - and those are filled when image was generated.
